packages/ui/src/components/utils/flex.styled.ts

```diff
@@ -1,56 +1,71 @@
 import { ReactNode } from 'react';
 import { css } from 'styled-components';
+import { MediaQuery } from '../../types/types';
+import { generateCss } from './generateCss';
+
+type JustifyType =
+  | 'flex-start'
+  | 'center'
+  | 'space-around'
+  | 'space-between'
+  | 'flex-end'
+  | 'evenly';
+
+type AlignType = 'flex-start' | 'center' | 'baseline' | 'flex-end' | 'stretch';
+
+type DirectionType = 'row' | 'column';
 
 export interface FlexProps {
   children?: ReactNode;
-  wrapReverse?: boolean;
-  noWrap?: boolean;
-  justify?:
-    | 'flex-start'
-    | 'center'
-    | 'space-around'
-    | 'space-between'
-    | 'flex-end'
-    | 'evenly';
-  align?: 'flex-start' | 'center' | 'baseline' | 'flex-end' | 'stretch';
-  content?:
-    | 'flex-start'
-    | 'flex-end'
-    | 'center'
-    | 'space-between'
-    | 'space-around';
-  direction?: 'row' | 'column';
-  gap?: number;
-  grow?: number;
+  wrapReverse?: MediaQuery<boolean>;
+  noWrap?: MediaQuery<boolean>;
+  justify?: MediaQuery<JustifyType>;
+  align?: MediaQuery<AlignType>;
+  direction?: MediaQuery<DirectionType>;
+  gap?: MediaQuery<number>;
+  grow?: MediaQuery<number>;
 }
 
 const justifyContent = css<FlexProps>`
-  ${props => props.justify && `justify-content: ${props.justify};`}
+  ${props =>
+    props.justify &&
+    generateCss(
+      ['justify-content'],
+      (item: JustifyType) => `${item}`,
+      props.justify,
+    )}
 `;
 
 const align = css<FlexProps>`
-  ${props => props.align && `align-items: ${props.align};`}
+  ${props =>
+    props.align &&
+    generateCss(['align-items'], (item: AlignType) => `${item}`, props.align)}
 `;
 
 const direction = css<FlexProps>`
-  ${props => props.direction && `flex-direction: ${props.direction};`}
+  ${props =>
+    props.direction &&
+    generateCss(
+      ['flex-direction'],
+      (item: DirectionType) => item,
+      props.direction,
+    )}
 `;
 
 const gap = css<FlexProps>`
-  ${props => props.gap && `gap: ${props.gap}px;`}
+  ${props =>
+    props.gap && generateCss(['gap'], (item: number) => `${item}px`, props.gap)}
 `;
 const grow = css<FlexProps>`
-  ${props => props.grow && `flex-grow: ${props.grow};`}
-`;
-
-const content = css<FlexProps>`
-  ${props => props.content && `align-content: ${props.content};`}
+  ${props =>
+    props.grow &&
+    generateCss(['flex-grow'], (item: number) => `${item}`, props.grow)}
 `;
 
 export const flex = css<FlexProps>`
+  display: flex;
```

Comment:
Don't use default css in utils

Reply:
The `flex` property defines that the property has `display: flex;` in it. If the default CSS should not be there then, we have to remove the `flex` utility.

Reply:
If we write `display: flex` here, it means all the components will be `flex` by default wherever you are using `flex` util. Irrespective of what props are present on the component.